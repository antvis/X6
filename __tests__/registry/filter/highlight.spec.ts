import { describe, expect, it } from 'vitest'
import {
  type HighlightArgs,
  highlight,
} from '../../../src/registry/filter/highlight'

describe('highlight filter', () => {
  it('should return default filter when no args provided', () => {
    const result = highlight()
    expect(result).toContain('flood-color="red"')
    expect(result).toContain('flood-opacity="1"')
    expect(result).toContain('radius="1"')
    expect(result).toContain('stdDeviation="0"')
    expect(result).toMatchSnapshot()
  })

  it('should apply custom color', () => {
    const args: HighlightArgs = { color: 'blue' }
    const result = highlight(args)
    expect(result).toContain('flood-color="blue"')
    expect(result).toContain('flood-opacity="1"')
  })

  it('should apply custom blur', () => {
    const args: HighlightArgs = { blur: 5 }
    const result = highlight(args)
    expect(result).toContain('stdDeviation="5"')
  })

  it('should apply custom width', () => {
    const args: HighlightArgs = { width: 3 }
    const result = highlight(args)
    expect(result).toContain('radius="3"')
  })

  it('should apply custom opacity', () => {
    const args: HighlightArgs = { opacity: 0.5 }
    const result = highlight(args)
    expect(result).toContain('flood-opacity="0.5"')
  })

  it('should apply multiple custom parameters', () => {
    const args: HighlightArgs = {
      color: 'green',
      blur: 2,
      width: 4,
      opacity: 0.8,
    }
    const result = highlight(args)
    expect(result).toContain('flood-color="green"')
    expect(result).toContain('stdDeviation="2"')
    expect(result).toContain('radius="4"')
    expect(result).toContain('flood-opacity="0.8"')
  })

  it('should handle edge cases for numeric values', () => {
    const args: HighlightArgs = {
      blur: 0,
      width: 0,
      opacity: 0,
    }
    const result = highlight(args)
    expect(result).toContain('stdDeviation="0"')
    expect(result).toContain('radius="0"')
    expect(result).toContain('flood-opacity="0"')
  })

  it('should generate valid SVG filter syntax', () => {
    const result = highlight()
    expect(result).toContain('<filter>')
    expect(result).toContain('</filter>')
    expect(result).toContain('<feFlood')
    expect(result).toContain('<feMorphology')
    expect(result).toContain('<feComposite')
    expect(result).toContain('<feGaussianBlur')
    expect(result).toContain('<feBlend')
  })
})
