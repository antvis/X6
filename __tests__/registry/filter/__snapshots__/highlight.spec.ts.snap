// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`highlight filter > should return default filter when no args provided 1`] = `
"<filter>
        <feFlood flood-color="red" flood-opacity="1" result="colored"/>
        <feMorphology result="morphed" in="SourceGraphic" operator="dilate" radius="1"/>
        <feComposite result="composed" in="colored" in2="morphed" operator="in"/>
        <feGaussianBlur result="blured" in="composed" stdDeviation="0"/>
        <feBlend in="SourceGraphic" in2="blured" mode="normal"/>
      </filter>"
`;
